app/services/foreman_bootdisk/renderer.rb

```diff
@@ -4,16 +4,35 @@ module ForemanBootdisk
   class Renderer
     include ::Foreman::Renderer
     include Rails.application.routes.url_helpers
+    delegate :logger, :to => :ForemanBootdisk
 
-    def generic_template_render
+    def generic_template_render(subnet = nil)
       tmpl = ProvisioningTemplate.find_by_name(Setting[:bootdisk_generic_host_template]) || raise(::Foreman::Exception.new(N_('Unable to find template specified by %s setting'), 'bootdisk_generic_host_template'))
-      @host = Struct.new(:token, :subnet).new(nil, nil)
-      unattended_render(tmpl.template)
+
+      if subnet.present?
+        # rendering a subnet-level bootdisk requires tricking the renderer into thinking it has a
+        # valid host, with a token, and with a tftp proxy
+        @host = Struct.new(:token, :subnet).new(
```

Comment:
i wasn't sure as it was not explicit in the new tests, but do you cover this usage case with a test?